src/MySqlConnector/MySqlDecimal.cs

```diff
@@ -0,0 +1,104 @@
+using System;
+using System.Globalization;
+
+namespace MySqlConnector;
+
+	public struct MySqlDecimal : IComparable, IComparable<MySqlDecimal>, IConvertible, IEquatable<MySqlDecimal>
+	{
+		private readonly string value;
+
+		internal MySqlDecimal(string val)
+		{
+			if (val.Length > 65)
+			{
+				throw new MySqlConversionException("Value is too large for Conversion");
+			}
+			value = val;
+		}
+
+		public decimal Value => Convert.ToDecimal(value, CultureInfo.InvariantCulture);
+
+		public int CompareTo(object? obj) => obj is MySqlDecimal other ?
+			((IComparable<MySqlDecimal>) this).CompareTo(other) :
+			throw new ArgumentException("CompareTo can only be called with another MySqlDateTime", nameof(obj));
+
+		public int CompareTo(MySqlDecimal other)
+		{
+			if(other.value == value)
+			{
+				return 1;
+			}
+			else
+			{
+				return -1;
+			}
+		}
```

Comment:
`CompareTo` needs to return `0` if both values are equal; otherwise, it needs to return a complete ordering over all values. 

I don't think it's necessary to implement `IComparable<T>`; it wasn't in the API I suggested here: https://github.com/mysql-net/MySqlConnector/issues/1070#issuecomment-952193539

Reply:
I am removing implementation of IComparable<T> interface all together now

Reply:
I am removing implementation of IComparable interface all together now,
Thanks for the review.